src/senseglove/senseglove_finger_distance/src/senseglove_finger_distance/finger_distance_node.py

```diff
@@ -16,7 +17,48 @@ def __init__(self, glove_nr="1", calib_mode='nothing', finger_nrs=[3, 7, 11, 15,
                          callback=self.callback, queue_size=1)  # queue size is necessary otherwise it is infinite
         self.pub = rospy.Publisher(self.senseglove_ns + "/finger_distances", FingerDistanceFloats, queue_size=1)
 
-        self.calibration = Calibration("default")
+        self._setup_calibration()
+
+    def _setup_calibration(self):
+        print("Setting up calibration")
+        # Calibration: setup default
+        self.calibration = Calibration(name="default")
+
+        # If calibration already on param server, load it
+        if rospy.has_param('~pinch_calibration_min') and rospy.has_param('~pinch_calibration_max'):
+            print("Found calibration data on server")
+            self.calibration = Calibration("from_param_server")
+            self.calibration.pinch_calibration_min = rospy.get_param('~pinch_calibration_min')
+            self.calibration.pinch_calibration_max = rospy.get_param('~pinch_calibration_max')
+        else:
+            print("No calibration data found, using defaults")
+
+        # Declare calibration service
+        self.calib_srv = rospy.Service("~Calibrate", Calibrate, self.calibrate_service)
+        self.calibrating = False
+        print(self.calib_srv.resolved_name)
+        print("Done setting up calibration")
+
+    def calibrate_service(self, call):
+        # Stop publishing commands & feedback
+        print("Executing calibration service")
+        self.calibrating = True
+
+        old_calib = self.calibration
+        if call.name is None:
+            call.name = "Unnamed_" + str(rospy.get_time())
+        print("calibration from call.name: ", call.name)
+        glove_nr = 0 if self.ns == '/lh/' else 1
+        self.calibration = Calibration(name=call.name, glove_nr=glove_nr)
+        result = self.calibration.run_interactive_calibration()
+
+        if not result:
+            # Failed; Restore previous calibration
```

Comment:
perhaps add user feedback here too

Reply:
I will add a print statement to give this feedback